go/src/lib.rs

```diff
@@ -170,11 +206,180 @@ pub unsafe extern "C" fn close_client(client_adapter_ptr: *const c_void) {
 pub unsafe extern "C" fn free_connection_response(
     connection_response_ptr: *mut ConnectionResponse,
 ) {
-    assert!(connection_response_ptr.is_null());
+    assert!(!connection_response_ptr.is_null());
     let connection_response = unsafe { Box::from_raw(connection_response_ptr) };
     let connection_error_message = connection_response.connection_error_message;
     drop(connection_response);
     if !connection_error_message.is_null() {
         drop(unsafe { CString::from_raw(connection_error_message as *mut c_char) });
     }
 }
+
+/// Deallocates a `CommandResponse`.
+///
+/// This function also frees the contained string_value and array_value. If the string_value and array_value are null pointers, the function returns and only the `CommandResponse` is freed.
+///
+/// # Panics
+///
+/// This function panics when called with a null `CommandResponse` pointer.
+///
+/// # Safety
+///
+/// * `free_command_response` can only be called once per `CommandResponse`. Calling it twice is undefined behavior, since the address will be freed twice.
+/// * `command_response_ptr` must be obtained from the `CommandResponse` returned in [`success_callback`] from [`command`].
+/// * `command_response_ptr` must be valid until `free_command_response` is called.
+/// * The contained `string_value` must be obtained from the `CommandResponse` returned in [`success_callback`] from [`command`].
+/// * The contained `string_value` must be valid until `free_command_response` is called and it must outlive the `CommandResponse` that contains it.
+#[no_mangle]
+pub unsafe extern "C" fn free_command_response(command_response_ptr: *mut CommandResponse) {
+    assert!(!command_response_ptr.is_null());
+    let command_response = unsafe { Box::from_raw(command_response_ptr) };
+    let string_value = command_response.string_value;
+    let string_value_len = command_response.string_value_len;
+    drop(command_response);
```

Comment:
This doesn't free the array. Please implement or leave a TODO.

Reply:
Added a TODO, will be handling it in the upcoming PR.